pkg/hook/controller/kubernetes_bindings_controller.go

```diff
@@ -50,6 +47,10 @@ type kubernetesBindingsController struct {
 
 	// dependencies
 	kubeEventsManager kube_events_manager.KubeEventsManager
+
+	// Snapshots cache for UpdateSnapshots
+	snapshotsCache      map[string][]ObjectAndFilterResult
```

Comment:
Why do we need a separate cache for snapshots? We already have a cache for filtered objects in the resource informer. Frankly, I do not understand the whole part about cached snapshots mode 😅  Cant we enable this mode by default without deferring and all staff?

Reply:
The "cached snapshots mode" is used only for the "Synchronization" phase. 

Binding context preparation for combined "Synchronization" binging contexts or for "Synchronization" with self-inclusion may require several calls to Snapshot() and objects can change between these calls. So the cache is introduced to ensure snapshot consistency.